File: Khelmianov-Artem/Term2/Block1/Filters/src/main/kotlin/bmp/Pixel.kt
package bmp


data class Pixel(
    var b: UByte = 0u,
    var g: UByte = 0u,
    var r: UByte = 0u
) {
    operator fun get(i: Int) = when (i) {
        0 -> b
        1 -> g
        2 -> r
        else -> throw IllegalArgumentException()
    }

    operator fun set(i: Int, v: UByte) = when (i) {
        0 -> b = v
        1 -> g = v
        2 -> r = v
        else -> {}
    }
}

File: DrumovMaxim/Semester2/Task1/Task1/Pixel.cs
namespace Task1
{
    public class Pixel
    {
        public byte[] ArrRGB = new byte[3];

        public Pixel(byte[] data)
        {
            ArrRGB[0] = data[0];
            ArrRGB[1] = data[1];
            ArrRGB[2] = data[2];
        }
    }
}
